packages/fixie/src/agent.ts

```diff
@@ -317,30 +345,213 @@ export class FixieAgent {
         config.public
       );
     }
+    return agent;
+  }
+
+  static spawnAgentProcess(agentPath: string, port: number): ChildProcess {
+    term(`🌱 Starting local agent process on port ${port}...\n`);
+    const pathToCheck = path.resolve(path.join(agentPath, 'dist', 'index.js'));
+    if (!fs.existsSync(pathToCheck)) {
+      throw Error(`Your agent was not found at ${pathToCheck}. Do you need to build your agent code first?`);
+    }
+
+    // Ensure we kill child process on exit.
+    const controller = new AbortController();
+    const { signal } = controller;
+    process.on('exit', () => {
+      console.log('🌱 Terminating agent process');
+      controller.abort();
+    });
```

Comment:
If we use https://www.npmjs.com/package/execa, this is handled for us automatically.